src/wallet/wallet.cpp

```diff
@@ -850,7 +852,7 @@ bool CWallet::GetAccountPubkey(CPubKey &pubKey, std::string strAccount, bool bFo
         if (!GetKeyFromPool(account.vchPubKey, false))
             return false;
 
-        SetAddressBook(account.vchPubKey.GetID(), strAccount, "receive");
+        SetAddressBook(AddDestinationForKey(account.vchPubKey, g_address_type), strAccount, "receive");
```

Comment:
Ugh, this sucks. We return a pubkey, so could end up adding for g_address_type and then getting used as g_change_type. Given this function is *only* used by GetAccountAddress in rpcwallet (which turns it into an address anyway), can you refactor a bit to make this return an account *address* instead?

Reply:
Will do.